reconcile/gitlab_housekeeping.py

```diff
@@ -37,13 +41,26 @@ def clean_pipelines(dry_run, gl_project, pipelines, pipeline_timeout=60):
 
         # pipeline_timeout converted in seconds
         if elapsed > pipeline_timeout * 60:
-            logging.info(['canceling', p['web_url']])
-            if not dry_run:
-                try:
-                    gl_project.pipelines.get(p['id']).cancel()
-                except gitlab.exceptions.GitlabPipelineCancelError as err:
-                    logging.error('unable to cancel {} - error message {}'
-                                  .format(p['web_url'], err.error_message))
+            timeout_pipelines.append(p)
+
+    return timeout_pipelines
+
+
+def clean_pipelines(dry_run: bool, gl_instance: str, gl_project_id: int,
+                    gl_settings: str, pipelines: List[str]) -> None:
+    if not dry_run:
+        gl_piplelines = GitLabApi(gl_instance,
+                                  project_id=gl_project_id,
+                                  settings=gl_settings).project.pipelines
+
+    for p in pipelines:
+        logging.info(['canceling', p['web_url']])
+        if not dry_run:
+            try:
+                gl_piplelines.get(p['id']).cancel()
+            except gitlab.exceptions.GitlabPipelineCancelError as err:
+                logging.error('unable to cancel {} - error message {}'
+                              .format(p['web_url'], err.error_message))
```

Comment:
Please, use f-strings
```suggestion
                logging.error(f'unable to cancel {p["web_url"]} - "
                       f"error message {err.error_message}')
```